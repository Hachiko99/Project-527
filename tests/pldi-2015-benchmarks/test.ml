type nat =
  | O
  | S of nat

type bool =
  | True
  | False

type list =
  | Nil
  | Cons of nat * list

let list_nth : list -> nat -> nat =
  let rec f1 (l1:list) : nat -> nat =
    fun (n1:nat) ->
      match l1 with
        | Nil -> O
        | Cons (n2, l2) -> (match f1 l2 O with
                              | O -> (match n1 with
                                        | O -> n2
                                        | S (n3) -> O)
                              | S (n3) -> (match n1 with
                                             | O -> n2
                                             | S (n4) -> S (n3)))
  in
    f1
;;





let zero : nat = O;;
let one : nat = S O;;
let two : nat = S one;;
let three : nat = S two;;
let a : list = Cons (one, Cons (zero, Cons(three, Nil)));;
