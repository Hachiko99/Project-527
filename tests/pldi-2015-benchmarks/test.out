(* Generated Mon Apr 11 18:44:34 2016 *)
An error was raised by this test: 2
Fatal error: exception Parser.Error
Raised at file "src/parser.ml", line 2754, characters 14-18
Called from file "src/synml.ml", line 138, characters 29-41
Called from file "src/synml.ml", line 143, characters 38-45
